### PR TITLE
docs: Add Portability section for TryFromBytes enum usage

### LINKED ISSUE #1778: Document how to implement enums with endian-specific tags
https://github.com/google/zerocopy/discussions/1288#discussioncomment-10799597

### PR DESCRIPTION
**Pull request by Aditya-PS-05**
closes #1778

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1182,6 +1182,27 @@ pub unsafe trait Immutable {
...
+///
+/// To ensure consistent endianness for enums with multi-byte representations,
+/// explicitly specify and convert each discriminant:
+///
+/// Here, `DataStoreVersion` enforces little-endian encoding for each `u32`
+/// discriminant, ensuring portability across platforms with differing native endianness.
+///
+/// ```
+/// # use zerocopy::{TryFromBytes, KnownLayout, IntoBytes, Immutable};
+/// #[derive(Debug, KnownLayout, Immutable, IntoBytes, TryFromBytes)]
```
[the remark this answers is not in the document]
Comment (Aditya-PS-05) [reply src/lib.rs:1195]: Ohh, I thought it relates to example. sorry for that. I am going to change it.

File: src/lib.rs [hunk 2]
```diff
@@ -1182,6 +1182,27 @@ pub unsafe trait Immutable {
...
+/// #[repr(u32)]
+/// pub enum DataStoreVersion {
+///     /// Version 1 of the data store.
+///     V1 = 9u32.to_le(),
+///
+///     /// Version 2 of the data store.
+///     V2 = 10u32.to_le(),
+/// }
+/// ```
+///
```
Comment (jswrenn) [inline src/lib.rs:1205]: Tightening up the prose a bit:
```suggestion
/// # Portability
///
/// To ensure consistent endianness for enums with multi-byte representations,
/// explicitly specify and convert each discriminant using `.to_le()` or
/// `.to_be()`; e.g.:
///
/// ```
/// # use zerocopy_derive::TryFromBytes;
/// // `DataStoreVersion` is encoded in little-endian.
/// #[derive(TryFromBytes)]
/// #[repr(u32)]
/// pub enum DataStoreVersion {
///     /// Version 1 of the data store.
///     V1 = 9u32.to_le(),
///
///     /// Version 2 of the data store.
///     V2 = 10u32.to_le(),
/// }
/// ```
///
```

**Review by jswrenn — CHANGES_REQUESTED** [review]
Just one last nit! Otherwise LGTM!

File: src/lib.rs (hunk 2 above)
Comment (Aditya-PS-05) [reply src/lib.rs]: Changes done

**Review by jswrenn — APPROVED** [review]

**Comment by joshlf** [comment]
Backporting in #2390

**Pull request merged**